org.civicrm.reqangular/src/common/services/api/contract.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/apis',
+  'common/services/api'
+], function (apis, _) {
+  'use strict';
+
+  apis.factory('api.contract', ['$log', 'api', function ($log, api) {
+    $log.debug('api.contract');
+
+    return api.extend({
+
+      /**
+       * This method returns all the contracts.
+       *
+       * It chains an additional call to the `getfulldetails` endpoint to also return
+       * full details on contracts
+       *
+       * @param  {Object} params  matches the api endpoint params (ex. contact_id)
+       * @return {Promise}
+       */
+      all: function (params) {
+        $log.debug('ContractAPI.all');
+
+        params = params || {};
+
+        params['api.HRJobContract.getfulldetails'] = {'jobcontract_id': '$value.id'};
```

Comment:
Yes agreed. But still I have 1 concern, the fulldetails API returns a lot of data, including `health`, `pension` etc. We dont need to get all those if we are not using it, it will hamper the performance. So please find out a way to get only `leave` data. (You might need to discuss with @davialexandre or @tunbola to know how to achieve this)

Now as this is a generic API we should send which field we want to fetch from parameters. Unless any specific field is mentioned all the data will be fetched. 

```
// CiviCRM API supports sending return param, where you can send the field you want to get,
but I am not sure how you can achieve the same in a chained API call so Please discuss with Davi or Tunbola
// This the how the code might look like
ContractAPI.all({ return: ['leave'] })